File: WEB-INF/classes/ClientHome.java
/*
 *Bodacious Assignment ** uMail **
 *
 *@author Parth Khandelwal
 *
 *Servlet to display Client's Home Screen
 */
import javax.servlet.*;
import javax.servlet.http.*;
import java.sql.*;
import java.io.*;

public class ClientHome extends HttpServlet {
	
	public void doGet(HttpServletRequest request, HttpServletResponse response) {
		try{
System.out.println("\nIn Home's Starting");		
			HttpSession session = request.getSession(true);
			String userid = (String)session.getAttribute("userid");
			System.out.println("User's id="+userid);
			ServletContext context = getServletContext();
			if(userid == null) {
				response.sendRedirect("/uMail/");
			} else {
				if(session.getAttribute("tag")==null) {
					session.setAttribute("tag","inbox");
				}
				String tag = (String)session.getAttribute("tag");
				response.setContentType("text/html");
				PrintWriter out = response.getWriter();
				out.println(""
				+"	<!doctype html>"
				+"	<html lang='en'>"
				+"	<head>"
				+"		<title>uMail</title>"
				+"		<meta charset='utf-8' />"
				+"		<style> li[value='"+tag+"'] {"
				+"			color				: black;"
				+"			font-weight			: bold;"
				+"			}"	
				+"		</style>"
				+"		<link rel='StyleSheet' type='text/css' href='css/clientHome.css' />"
				+"		<script src='javascript/clientHome.js'> </script>"
				+"	</head>"
				+"	<body>"
				+"		<div id='mainDiv'>"
				+"			<header id='mainHeader'>"
				+"				<div id='titleDiv'>"
				+"					<h2 id='title'> uMail </h2>"
				+"				</div>"
				+"				<div id='logoutDiv'>"
				+"					<input type='button' id='logoutBtn' name='logout' value='Logout' onclick='logMeOut()' />"
				+"				</div>"
				+"			</header>"
				+"			<div id='centerDiv'>"
				+"				<nav id='options'>"
				+"					<input type='button' id='compose' name='compose' value='Compose' onclick='showComposeDiv()' />"
				+"					<input type='hidden' id='selectedOption' value='inbox' />"
				+"					<ul>"
				+"						<li name='option' value='inbox' onClick='optionClicked(\"inbox\")'>Inbox</li>"
				+"						<li name='option' value='sentmail' onClick='optionClicked(\"sentmail\")'>Sent Mail</li>"
				+"						<li name='option' value='drafts' onClick='optionClicked(\"drafts\")'>Drafts</li>"
				+"						<li name='option' value='outbox' onClick='optionClicked(\"outbox\")'>Outbox</li>"
				+"						<li name='option' value='trash' onClick='optionClicked(\"trash\")'>Trash</li>"
				+"					</ul>"
				+"				</nav>"
				+"				<section id='mainSection'>"
				+"					<div id='mailsDiv'>");
System.out.println("\n***Initiating LoadMails***");			
					RequestDispatcher reqDis = context.getRequestDispatcher("/LoadMails");
					reqDis.include(request, response);
System.out.println("\n***#Completed Loading#***");			
					out.println(""
+"				</div>"
+"					<div id='mailDetailDiv'>"
+"						<header id='toolbar'>"
+"							<input type='button' id='back' name='back' value='Back' onclick='hideMailDetail()'/>"
+"							&nbsp;&nbsp;&nbsp;&nbsp;"
+"							<input type='button' id='reply' name='reply' value='Reply' onclick='replyToMail()' />"
+"							<input type='button' id='forward' name='forward' value='Forward' onclick='forwardMail()' />"
+"						</header>"
+"						<font size='1.1em'><b>From &nbsp;&nbsp;&nbsp;&nbsp;&nbsp; :</b> &nbsp;<span id='from'></span> </br>"
+"						<b>To &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; :</b> &nbsp;<span id='to'></span> </br>"
+"						<b>Subject &nbsp; :</b> &nbsp;<span id='subject'></span> </br> </br>"
+"						</font>"
+"						<textarea name='messageTA' id='msgContent'>"
+"						</textarea>"
+"					</div>"
+"				</section>"
+"			</div>"
+"		</div>"
+"		<div id='composeMailRootDiv' onClick='outerClicked(window.event)'>"
+"			<div id='composeMailCenterDiv' onmouseover='darkenBackground()' onmouseout='blurBackground()' >"
+"				<form method='post' id='composeForm' >"
+"					<font color='cyan' size='1.1em'>To: &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<input type='email' name='to' id='composeTo' maxlength='30' placeholder='A User ID' required onfocus='darkenBackground()' /> </br>"
+"					Cc: &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<input type='email' name='cc' id='composeCc' placeholder='User-ID seperated with a Whitespace' onfocus='darkenBackground()' multiple /> </br>"
+"					Bcc: &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<input type='email' name='bcc' id='composeBcc' placeholder='User-ID seperated with a Whitespace' onfocus='darkenBackground()' multiple /> </br>"
+"					Subject: <input type='text' name='subject' id='composeSubject' placeholder='Mandatory To Fill' required onfocus='darkenBackground()' /> </br> </br>"
+"					</font>"
+"					<textarea name='content' id='composeContent' placeholder='Content' onfocus='darkenBackground()'></textarea> </br>"
+"					<input type='submit' id='sendBtn' name='sendBtn' value='Send' formaction='SendMail' />"
+"					<input type='submit' id='saveBtn' name='saveBtn' value='Save' formaction='SaveMail' />"
+"					<input type='button' value='Discard' onClick='closeComposeMail()' />"
+"				</form>"
+"			</div>"
+"		</div>"
+"	</body>"
+"</html>");

			out.close();					
		}					
	}
		catch(Exception e){
			e.printStackTrace();
			System.out.println("Exceptionnnnnnnn ");
		}
	}
	
	public void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
		doGet(request, response);
	}
}

File: javascript/clientHome.js
// uMail JS File for Client's Home Page

function logMeOut() {
	document.location.href='Logout?';
}

function selectAllMails() {
	var mailArr = document.querySelectorAll('#check');
	if(document.querySelector('#selectAll').checked){
		for(var i=0; i<mailArr.length; i++) {
			mailArr[i].checked = true;
		}
		setToolbarBtnState('inline');
	} else {
		for(var i=0; i<mailArr.length; i++) {
			mailArr[i].checked = false;
		}
		setToolbarBtnState('none');
	}
}

function showMailDetail(id) {
	document.querySelector('#mailsDiv').style.display = 'none';
	document.querySelector('#mailDetailDiv').style.display = 'block';
	
	var mailDetail = document.getElementsByName(id);
	
	document.getElementById('from').textContent = mailDetail[0].textContent;
	document.getElementById('to').textContent = mailDetail[1].textContent;
	document.getElementById('subject').textContent = mailDetail[2].textContent;
	document.getElementById('msgContent').value = mailDetail[4].value;
}

function hideMailDetail() {
	document.querySelector('#mailDetailDiv').style.display = 'none';
	document.querySelector('#mailsDiv').style.display = 'block';
}

function showComposeDiv() {
	document.querySelector('#composeMailRootDiv').style.display = '-webkit-box';
}

function closeComposeMail() {
	document.querySelector('#composeMailRootDiv').style.display = 'none';
	var fields = document.querySelectorAll('input[type="text"], input[type="email"]');
	for(var i=0; i<fields.length; i++) {
		fields[i].value = '';
	}
	document.querySelector('#composeContent').value = '';
}

function outerClicked(event){
	if(event.target == event.currentTarget)
		closeComposeMail();
}

// function enableToggleToolbarButton() {
	// var checks = document.querySelectorAll('input[name="check"]');
	// for(var i=0; i<checks.length; i++) {
		// checks[i].onchange = checkStateToDisplayToolbarButton;
	// }
// }

function checkStateToDisplayToolbarButton() {
	var checks = document.querySelectorAll('input[name="check"]');
	var i=0;
	var displayType = 'inline';
	for(i; i<checks.length; i++) {
		if(checks[i].checked) {
			break;
		}
	}
	if(i==checks.length) {
		displayType = 'none';
		document.querySelector('#selectAll').checked = false;
	}
	setToolbarBtnState(displayType);
}

function setToolbarBtnState(displayType) {
	var toolbarBtn = document.querySelectorAll('input[form="mailForm"]');
	for(var j=0; j<toolbarBtn.length; j++) {
		toolbarBtn[j].style.display = displayType;
	}
}

/* function enableOptionClick() {
	var options = document.querySelectorAll('li[name="option"]');
	for (var i=0; i<options.length; i++) {
		options[i].onclick = optionClicked(options[i]);
	}
} */

function optionClicked(option) {
	var options = document.querySelectorAll('li');
	for (var i=0; i<options.length; i++) {
		options[i].style.color = '#fd7214';
		options[i].style.fontWeight = 'normal';
	}
	var selected = document.querySelector('li[value=' +option+ ']');
	selected.style.color = 'black';
	selected.style.fontWeight = 'bold';
	document.getElementById('selectedOption').value = option;
	
	var request = new XMLHttpRequest();
	
	request.onreadystatechange = function (){
		if(request.readyState == 0 && request.status==200){
			document.getElementById('mailsDiv').innerHTML = '<center><h1>Connecting To Server</h1><center>';
		} else if(request.readyState == 1 && request.status==200 ){
			document.getElementById('mailsDiv').innerHTML = '<center><h1>Connection Established</h1><center>';
		} else if(request.readyState == 2 && request.status==200){
			document.getElementById('mailsDiv').innerHTML = '<center><h1>Retrieving Data</h1><center>';
		} else if(request.readyState == 3 && request.status==200){
			document.getElementById('mailsDiv').innerHTML = '<center><h1>Processing Data</h1><center>';
		} else if(request.readyState == 4 && request.status==200){
			document.getElementById('mailsDiv').innerHTML = request.responseText;
		} else if(request.status == 500) {
			document.getElementById('mailsDiv').innerHTML = '<center><h1>Server Error</h1><center>';
		} else {
			document.getElementById('mailsDiv').innerHTML = '<center><p>Loading...<p><center>';
		}
	};
	request.open('get','LoadMails?option=' +option ,true);
	request.send(null);
	hideMailDetail();
}

function replyToMail() {
	document.getElementById('composeTo').value = document.getElementById('from').textContent;
	document.getElementById('composeSubject').value = document.getElementById('subject').textContent;
	showComposeDiv();
}

function forwardMail() {
	document.getElementById('composeSubject').value = document.getElementById('subject').textContent;
	document.getElementById('composeContent').value = document.getElementById('msgContent').value;
//	document.getElementById('composeContent').value = document.getElementById('msgContent').value;
	showComposeDiv();
}
function blurBackground() {
	document.querySelector('#composeMailRootDiv').style.backgroundColor = 'rgba(0,0,0,0.7)';
	//alert('here');
}

function darkenBackground() {
	document.querySelector('#composeMailRootDiv').style.backgroundColor = 'rgba(0,0,0,0.86)';
	//alert('here');
}
